api/src/lib.rs

```diff
@@ -567,15 +579,17 @@ impl Client {
 
     fn delete<LocationT, ErrorT>(&self, url: LocationT) -> Result<()>
     where
-        LocationT: IntoUrl + Display,
+        LocationT: IntoUrl + Display + Clone,
         for<'de> ErrorT: Deserialize<'de> + ApiError,
     {
         debug!("Attempting DELETE `{}`", url);
         let http_response = self
-            .http_client
-            .delete(url)
-            .headers(self.headers.clone())
-            .send()
+            .handle_timeouts(|| {
+                self.http_client
+                    .delete(url.clone())
+                    .headers(self.headers.clone())
+                    .send()
+            })
```

Comment:
This will fail with a 404 if the first timed out request had actually succeded (and the timeout was on the response side). Not sure how to handle non-idempotent requests---we could just not wrap them or explicitly succeed follow up requests on 404 for deletion (this would have to make the `handle_timeouts` a leakier abstraction)